src/encoded/ingestion/loadxl_extensions.py

```diff
@@ -0,0 +1,75 @@
+import re
+from typing import Generator, Optional
+from dcicutils.misc_utils import VirtualApp
+from snovault.loadxl import load_all_gen as loadxl_load_data
+from .submission_folio import SmahtSubmissionFolio
+
+
+def load_data_into_database(data: dict[str, list[dict]], portal_vapp: VirtualApp, validate_only: bool = False) -> dict:
+
+    def package_loadxl_response(loadxl_response: Generator[bytes, None, None]) -> dict:
+        LOADXL_RESPONSE_PATTERN = re.compile(r"^([A-Z]+):\s*(.*)$")
+        LOADXL_ACTION_NAME = {"POST": "created", "PATCH": "updated", "SKIP": "skipped", "CHECK": "validated", "ERROR": "errors"}
+        response = {value: [] for value in LOADXL_ACTION_NAME.values()}
+        for item in loadxl_response:
+            # ASSUME each item in the loadxl response looks something like one of (string or bytes):
+            # POST: beefcafe-01ce-4e61-be5d-cd04401dff29
+            # PATCH: deadbabe-7b4f-4923-824b-d0864a689bb
+            # SKIP: feedbeef-eb17-4406-adb8-060ea2ae2180
+            # CHECK: cafebabe-eb17-4406-adb8-0eacafebabe
+            # ERROR: deadbeef-483e-4a08-96b9-3ce85ce8bf8c
+            # Note that SKIP means skip POST (create); it still may do PATCH (update), if overwrite.
+            if isinstance(item, bytes):
+                item = item.decode("ascii")
+            elif not isinstance(item, str):
```

Comment:
I think I would like a toplevel function like (untested):
```
def maybe_decode_ascii(str_or_bytes):
    if isinstance(str_or_bytes, bytes):
        return item.decode('ascii')
    return str_or_bytes
```
so you can here do:
```suggestion
            item = maybe_decode_ascii(item)
            if not isinstance(item, str):
```

or the function could be:
```
def maybe_decode_bytes(str_or_bytes, *, encoding='utf-8'):
    if isinstance(str_or_bytes, bytes):
        return item.decode(encoding)
    return str_or_bytes
```
so that the call can be
```
            item = maybe_decode_bytes(item, encoding='ascii')
            if not isinstance(item, str):
```

Reply:
Done